### ISSUE TITLE
SynoKernel not working on 7.2

### ISSUE THREAD

**Issue by surge919** [SynoCommunity/spksrc#5772]
### Is this a new Bug?

- [X] I checked that the bug hasn't been reported before

### Package Name

SynoKernel USB Serial Drivers

### Package Version

1.1-8

### Device Model

DS918+

### Device Architecture

x86_64

### Firmware Version

7.2

### What happened?

I upgraded my DS918+ from 7.1 to 7.2
USB no longer works and I've confirmed I have the latest package

checking the modules in
/volume1/@appstore/synokernel-usbserial/lib/modules/
I see the following apollolake-7.0/4.4.180

uname shows it uses 4.4.302+

Linux OurStuff 4.4.302+ #64570 SMP Sat Jun 3 00:09:29 CST 2023 x86_64 GNU/Linux synology_apollolake_918+

Even though it doesn't match, I tried it anyways by copying usbserial.ko to /lib/modules and get the following error

sudo insmod /lib/modules/usbserial.ko
insmod: ERROR: could not insert module /lib/modules/usbserial.ko: Invalid module format

### Reproduction steps

1.  download latest package
2.  copied module to /lib/modules
3.  sudo insmod /lib/modules/usbserial.ko
insmod: ERROR: could not insert module /lib/modules/usbserial.ko: Invalid module format

### Install Log

```log
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin preinst
2023/06/10 21:44:12	Begin load_variables_from_file
2023/06/10 21:44:12	End load_variables_from_file
2023/06/10 21:44:12	Begin initialize_variables
2023/06/10 21:44:12	End initialize_variables
2023/06/10 21:44:12	===> Step preinst. STATUS=INSTALL USER= GROUP= SHARE_PATH=
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End preinst ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mkdir -p /volume1/@appstore/synokernel-usbserial -m 755
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mkdir -p /volume1/@appstore/synokernel-usbserial -m 755 ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/rm -rf /volume1/@appstore/synokernel-usbserial
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/rm -rf /volume1/@appstore/synokernel-usbserial ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/package /volume1/@appstore/synokernel-usbserial
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/package /volume1/@appstore/synokernel-usbserial ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/rm -rf /var/packages/synokernel-usbserial
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/rm -rf /var/packages/synokernel-usbserial ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mkdir -p /var/packages/synokernel-usbserial -m 755
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mkdir -p /var/packages/synokernel-usbserial -m 755 ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/touch /var/packages/synokernel-usbserial/installing
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/touch /var/packages/synokernel-usbserial/installing ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/INFO /var/packages/synokernel-usbserial/INFO
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/INFO /var/packages/synokernel-usbserial/INFO ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/rm -rf /var/packages/synokernel-usbserial/scripts
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/rm -rf /var/packages/synokernel-usbserial/scripts ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/scripts /var/packages/synokernel-usbserial/scripts
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/scripts /var/packages/synokernel-usbserial/scripts ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/rm -rf /var/packages/synokernel-usbserial/WIZARD_UIFILES
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/rm -rf /var/packages/synokernel-usbserial/WIZARD_UIFILES ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/WIZARD_UIFILES /var/packages/synokernel-usbserial/WIZARD_UIFILES
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/WIZARD_UIFILES /var/packages/synokernel-usbserial/WIZARD_UIFILES ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/rm -rf /var/packages/synokernel-usbserial/conf
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/rm -rf /var/packages/synokernel-usbserial/conf ret=[0]
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/conf /var/packages/synokernel-usbserial/conf
2023/06/10 21:44:12	install synokernel-usbserial 1.1-8 End /bin/mv -f /volume1/@tmp/synopkg/install.0MIco3/conf /var/packages/synokernel-usbserial/conf ret=[0]
2023/06/10 21:44:15	install synokernel-usbserial 1.1-8 Begin postinst
2023/06/10 21:44:15	Begin load_variables_from_file
2023/06/10 21:44:15	End load_variables_from_file
2023/06/10 21:44:15	Begin initialize_variables
2023/06/10 21:44:15	End initialize_variables
2023/06/10 21:44:15	===> Step postinst. STATUS=INSTALL USER= GROUP= SHARE_PATH=
2023/06/10 21:44:15	Begin save_wizard_variables
2023/06/10 21:44:15	End save_wizard_variables
2023/06/10 21:44:15	Begin syno_sync_var_folder
2023/06/10 21:44:15	End syno_sync_var_folder
2023/06/10 21:44:15	install synokernel-usbserial 1.1-8 End postinst ret=[0]
2023/06/10 21:44:15	install synokernel-usbserial 1.1-8 Begin start-stop-status start
2023/06/10 21:44:15	install synokernel-usbserial 1.1-8 End start-stop-status start ret=[0]
```

**Comment by th0ma7** [comment]
Synology haven't yet provided linux kernel sources part of DSM-7.2.  Neither for 7.1 for that matter but it occurred that the version had not changed from 7.0.

You can request kernel sources using synology support (which I also did 2 weeks ago).

Eventually they will end-up here:
https://archive.synology.com/download/ToolChain/Synology%20NAS%20GPL%20Source

**Comment by surge919** [comment]
thanks for the update.. I'll watch this and the other post for any updates (if any).
The other thread doesn't seem hopeful, so does this seem like the end of USB support on Synology?

**Comment by jisakiel** [comment]
USB drivers do work as reported on the other thread. Synology will likely refuse support for any issues (or at least any issues they can attribute to the kernel), and it might become tighter locked in the future, but for now drivers keep working.

That said, spk needs updates to support 7.2 as the kernel version changed.

**Comment by th0ma7** [comment]
DSM-7.1.1 kernel sources are now out:
https://archive.synology.com/download/ToolChain/Synology%20NAS%20GPL%20Source

I doubt 7.2 sources will be available before next DSM version.  I'll update the framework and synokernel packages to reflect this in the coming weeks.  Still, 7.2 will be missing unless I "fake" the sources intentionally...  or synology decides to make it available as well.

**Comment by scrapix** [comment]
I've actually stumbled upon the same issue while trying to compile
`
MODULE=net/netfilter
MODULE6=net/ipv6/netfilter
LIB=lib
`
Also leads to 
`*.ko: Invalid module format` :(

**Comment by barndawgie** [comment]
I have a USB Z-Wave stick which no longer works after upgrade to DSM 7.2; I simply no loner get a /dev/ttyUSB device.

Is that likely due to this issue?

**Comment by bmedici** [comment]
Same here, SynolCli Kernel Tools is just not available on my DS923+ for now.

**Comment by th0ma7** [comment]
And sadly it won't be until Synology releases the kernel sources for dsm 7.2 neither.

**Comment by jisakiel** [comment]
For what is worth, I did request them the sources via ticket, only to be told that they'll release whenever 7.2 is made generally available. Which is a violation of the GPL, something I also pointed out to them. Waiting on the outcome of the Vizio case to press them further on it.

**Comment by scrapix** [comment]
> I have a USB Z-Wave stick which no longer works after upgrade to DSM 7.2; I simply no loner get a /dev/ttyUSB device.
> 
> Is that likely due to this issue?

In case you havent managed to run your USB Device on DSM 7.2 you should be able to do with following guide:
https://gist.github.com/scrapix/2f9ad27459c3646732f510f50ead3410

**Comment by MikeD-NL** [comment]
I've got something similar.
As a backup NAS I bought a DS220+ next to my 720+ on which it all works like a charm! (DSM 7.2-64570 Update 3 and Domoticz Version: 2021.1 build 14070)

I ran into problems with de latest version of Domoticz and my Aeotec ZWave usb stick and installed the Domoticz version of SynoCommunity (Version: 2020.2) on my DS220+ with DSM 7.2-64570 Update 1.

I above mentioned USB serial drivers and Domiticz sees a /dev/ttyACM0 when I select the "OpenZwave USB" in "Hardware/Type", but says "Manager down!" and is not working.

In the logging of Domoticz I see the next two lines:
```
2023-08-11 12:21:22.040 Status: OpenZWave: using config in: /volume1/@appstore/domoticz/bin/Config/
2023-08-11 12:21:22.040 Error: OpenZWave: Exception. Type: 1, Msg: Cannot Find Configuration Files, File: /spksrc/spk/domoticz/work-x64-7.1/openzwave-1.6.1914/cpp/src/Options.cpp (Line 90) OpenZWave.cpp:894 
```
